### PR TITLE
Fix insert extension test for local dev testing mode

### LINKED ISSUE #3363: test-insert-extension is failing locally
screen shot 2016-05-27 at 10 13 17 am

### PR DESCRIPTION
**Pull request by muxin**
Fixes #3363

### CODE REVIEW

File: src/insert-extension.js [hunk 1]
```diff
@@ -59,8 +59,10 @@ function createAmpExtensionScript(win, extension) {
   ampExtensionScript.setAttribute('custom-element', extension);
   ampExtensionScript.setAttribute('data-script', extension);
   const pathStr = win.location.pathname;
+  const useCompiledJs = window.ampTestRuntimeConfig ?
```
Comment (dvoytenko) [inline src/insert-extension.js]: This is certainly fine. But for "shortness" you can also do this:
```
const useCompiledJs = window.ampTestRuntimeConfig &&
    window.ampTestRuntimeConfig.useCompiledJs;
```

This is a bit more idiomatic...

File: src/insert-extension.js [hunk 2]
```diff
@@ -59,8 +59,10 @@ function createAmpExtensionScript(win, extension) {
...
+      window.ampTestRuntimeConfig.useCompiledJs : undefined;
```
Comment (dvoytenko) [inline src/insert-extension.js]: Should this be `win` or `window`?

Comment (muxin) [inline src/insert-extension.js]: `window.ampTestRuntimeConfig` is only assigned in testing mode as a global variable. So I guess it doesn't belong to the `win` which is the testing iframe.

File: src/insert-extension.js (hunk 1 above)
Comment (muxin) [inline src/insert-extension.js]: Done

**Comment by muxin** [comment]
@cramforce PTAL

**Comment by dvoytenko** [comment]
LGTM

File: src/insert-extension.js [hunk 3]
```diff
@@ -59,8 +59,10 @@ function createAmpExtensionScript(win, extension) {
   ampExtensionScript.setAttribute('custom-element', extension);
   ampExtensionScript.setAttribute('data-script', extension);
   const pathStr = win.location.pathname;
+  const useCompiledJs = window.ampTestRuntimeConfig &&
```
Comment (cramforce) [inline src/insert-extension.js]: Add `win.AMP_TEST` at the start. This will help with dead code elimination.

Comment (muxin) [inline src/insert-extension.js]: Done

**Pull request merged**